### PR TITLE
enable HTTP caching for HTML message bodies

### LINKED ISSUE #799: preload and cache HTML messages
To speed up opening HTML messages, we could preload them in background

As @LukasReschke said
> What I can think about as alternative is to somehow preload the pages and set regular HTTP caching headers on the HTML messages.

- [ ] cache HTML messages
- [ ] preload HTML messages

see #768, #480

### PR DESCRIPTION
**Pull request by ChristophWurst**
partially fixes #799

Firefox seems to ignore it for some reason, but Chrome caches message bodies as expected.

@LukasReschke @DeepDiver1975 @Xenopathic review please

### CODE REVIEW

File: lib/controller/messagescontroller.php [hunk 1]
```diff
@@ -184,6 +184,10 @@ public function getHtmlBody($messageId) {
 				$htmlResponse->setContentSecurityPolicy($policy);
 			}
 
+			// Enable caching
+			$htmlResponse->cacheFor(60 * 60);
+			$htmlResponse->addHeader('Pragma', 'cache');
```
Comment (LukasReschke) [inline lib/controller/messagescontroller.php]: Can we haz :cookie:  and unit tests for this? :smile:

Comment (ChristophWurst) [inline lib/controller/messagescontroller.php]: :cookie: ? :D
I'll write some tests once #750 is merged

Comment (LukasReschke) [inline lib/controller/messagescontroller.php]: GO GO GO GO GO GO GO - IT IS MERGED JIMMY! ;-) :cookie:!

File: tests/controller/messagescontrollertest.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ */
+
+use OCA\Mail\Controller\MessagesController;
+
+/**
+ * Tests for lib/controller/messagescontroller
+ *
+ * @author Christoph Wurst
+ */
+class messagescontrollertest extends \Test\TestCase {
```
Comment (LukasReschke) [inline tests/controller/messagescontrollertest.php]: That casing is interesting. :speak_no_evil:

File: tests/controller/messagescontrollertest.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+		$this->assertInstanceOf('\OCA\Mail\Http\HtmlResponse', $response);
+		$headers = $response->getHeaders();
+		// Check for header existense
+		$this->assertTrue(array_key_exists('Cache-Control', $headers));
+		$this->assertTrue(array_key_exists('Pragma', $headers));
+
+		// Check header values
+		$this->assertSame($headers['Cache-Control'], 'max-age=3600, must-revalidate');
+		$this->assertSame($headers['Pragma'], 'cache');
```
Comment (LukasReschke) [inline tests/controller/messagescontrollertest.php]: That looks kinda complicated. Can't we not just compare the response object?
```
$expectedResponse = new WhateverResponse(…);
$expectedResponse->myYolo = 'swag';
$this->assertEquals($expectedResponse, $response);
```

**Comment by ChristophWurst** [comment]
@LukasReschke @DeepDiver1975 @jancborchardt is there anything else needed? could anybody please test this? :-)

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**